asciinema/__main__.py

```diff
@@ -21,7 +21,7 @@ def positive_float(value):
 
 
 def rec_command(args, config):
-    api = Api(config.api_url, os.environ.get("USER"), config.api_token)
+    api = Api(config.api_url, os.environ.get("USER"), config.api_token) if config.record_upload else None
```

Comment:
What do you think about moving `upload` boolean up from `record.py` to `rec_command()` in `__main__.py`, and also generating a default filename (in tmp) there, then passing it down to `RecordCommand`. For example:

```python
def rec_command(args, config):
    api = Api(config.api_url, os.environ.get("USER"), config.api_token)
    if args.filename == "":
        filename = _tmp_path()
        upload = config.record_upload
    else:
        filename = args.filename
        upload = False
    return RecordCommand(api, filename, args.command, args.title, args.yes, args.quiet, args.max_wait, upload)
```